lib/privatekey.js

```diff
@@ -384,9 +385,27 @@ PrivateKey.prototype.toObject = PrivateKey.prototype.toJSON = function toObject
   }
 }
 
+/**
+ * Will return child PrivateKey derived from id by IBE.
+ *
+ * @returns {PrivateKey} A new valid instance of IBE child PrivateKey
+ */
+PrivateKey.prototype.childKey = function (id) {
```

Comment:
even though this method is short, i believe it should not be inside the main library. there should be a new folder, "idm", similar to "ecies", "message", "mnemonic" which is actually a new project inside this repo. it contains the methods related to IDM. the reason to do it that way is the main library is bitcoin stuff only - only standard things that were there in the beginning and are necessary for any bitcoin app. IDM is an optional add-on that only some apps will use, similar to ECIES, Message and Mnemonic.

Reply:
I got independent library for it.
https://github.com/monkeylord/bitcoin-ibe

The reason why I added in Private/PublicKey is because it can be chained and easy to use if we do so.
IBE is like HD, but is superior if you are developing a biticoin app. HD is only for wallets, and HD's key deriving path cannot be controlled by app.
However, IBE is app friendly.
Like:
  UserMasterKey.childKey('memo').childKey('comment').childKey('1')
So user can login with a single masterkey, and app key hierarchy is quite clear and easy to code.
(HD can do similar thing, but you have to map every key to your app's logic. It's not friendly unless you develop wallets)

I believe that if an app is multi-key, IBE is a necessary. So I added it into Private/Public Key.

Reply:
OK - makes sense. It is true that we already have BIP32 in the library. May as well add this. bsv is really primarily a wallet tool library, not an implementation of Bitcoin. The only catch here is that it's good if the main library remains small so it loads easy in a browser, but this code is pretty small.